src/test/groovy/lang/IntRangeTest.groovy

```diff
@@ -138,4 +138,15 @@ class IntRangeTest extends GroovyTestCase {
         assert bs[-1..<-7].toString() == '{0, 5}'
         assert bs[20..<-8].toString() == '{2, 3}'
     }
+
+    void testHashCode(){
```

Comment:
best place to test hashcode is HashMap. Let know if need to check this up for more range of values.